packages/amplication-server/src/core/UsageInsights/usageInsights.service.ts

```diff
@@ -0,0 +1,230 @@
+          GROUP BY year, month, time_group
+          ORDER BY year, month, time_group;
+        `;
+        break;
+      case EnumBlockType.PluginInstallation:
+        results = await this.prisma.$queryRaw`
+          SELECT DATE_PART('year', bv."updatedAt") as year, DATE_PART('month', bv."updatedAt") as month, DATE_PART('month', bv."updatedAt") as time_group, COUNT(bv.*) AS count
+          FROM "BlockVersion" bv
+          JOIN "Block" b ON bv."blockId" = b."id"
+          JOIN "Resource" r ON b."resourceId" = r."id"
+          WHERE b."blockType" = 'PluginInstallation'
+          AND r."projectId" IN (${Prisma.join(projectIds)})
+          AND bv."updatedAt" >= ${startDate} AND bv."updatedAt" <= ${endDate}
+          GROUP BY year, month, time_group
+          ORDER BY year, month, time_group;
+        `;
+        break;
+      default:
+        throw new Error(`Block type ${blockType} is not supported`);
+    }
+
+    return {
+      results: Object.values(this.translateToAnalyticsResults(results)),
+    };
+  }
+
+  async getUsageInsights(
+    args: BaseUsageInsightsArgs
+  ): Promise<UsageInsightsResult> {
+    const builds = await this.countBuilds(args);
+    const entities = await this.countEntityChanges(args);
+    const moduleActions = await this.countBlockChanges({
+      ...args,
+      blockType: EnumBlockType.ModuleAction,
+    });
+    const plugins = await this.countBlockChanges({
+      ...args,
+      blockType: EnumBlockType.PluginInstallation,
+    });
+
+    this.logger.debug("Usage insights result", {
+      builds,
+      entities,
+      moduleActions,
+      plugins,
+    });
+
+    return {
+      builds,
+      entities,
+      moduleActions,
+      plugins,
+    };
+  }
+
+  async getEvaluationInsights(
+    args: BaseUsageInsightsArgs
+  ): Promise<EvaluationInsights> {
+    const loc = await this.countLinesOfCode(args);
+    const timeSaved = await this.evaluateTimeSaved(loc);
+    const costSaved = await this.evaluateCostSaved(loc);
+    const codeQuality = await this.evaluateCodeQuality(loc);
+
+    this.logger.debug("Evaluation insights result", {
+      loc,
+      timeSaved,
+      costSaved,
+      codeQuality,
+    });
+
+    return {
+      loc,
+      timeSaved,
+      costSaved,
+      codeQuality,
+    };
+  }
+
+  private async evaluateTimeSaved(linesOfCode: number) {
+    const divisor = 10.41;
+    const timeSaved = linesOfCode / divisor;
+    return Math.round(timeSaved);
+  }
+
+  private async evaluateCostSaved(linesOfCode: number) {
+    const multiplier = 12;
+    const coastSaved = multiplier * linesOfCode;
+    return Math.round(coastSaved);
+  }
+
+  private async evaluateCodeQuality(linesOfCode: number) {
+    const multiplier = 14;
+    const divisor = 1000;
+    const bugsPrevented = (multiplier * linesOfCode) / divisor;
+
+    return Math.round(bugsPrevented);
+  }
+
+  private translateToAnalyticsResults(results: QueryRawResult[]) {
```

Comment:
Here again, why do we need this - is it just to rename time_group to timeGroup? Can we avoid the need for this translation?

Reply:
The parsing later is not only for the name, but also for type. Moreover, one of the inputs of the function named timeGroup